docs/branch-operations.md

```diff
@@ -0,0 +1,540 @@
+- **Atomic operations**: Changes either fully succeed or fully fail (no partial states)
+
+## Understanding Branch Divergence
+
+Before performing branch operations, it's helpful to understand how branches relate:
+
+```clojure
+;; Check relationship between branches
+(def divergence @(fluree/branch-divergence conn "ledger:feature" "ledger:main"))
+
+;; Returns:
+{:common-ancestor "fluree:commit:sha256:..."
+ :can-fast-forward true  ; or false
+ :fast-forward-direction :branch1-to-branch2}  ; or :branch2-to-branch1, nil
+;; Note: branch1-ahead and branch2-ahead are documented but not yet implemented
+```
+
+## Operations
+
+### 1. MERGE - Updates Target Branch with Source Changes
+
+Merges commits from source branch into target branch. Currently supports:
+- **Fast-forward** when target branch is behind source (no divergence)
+- **Squash** to combine all source commits into a single commit on target
+
+#### How Squash Works
+
+When using `squash? true`, Fluree:
+1. Collects all flakes (assertions/retractions) from source commits
+2. Groups them by `[subject predicate object datatype meta]`
+3. Counts net operations (assertions - retractions) for each unique value
+4. Produces final flakes:
+   - Net positive → assertion
+   - Net negative → retraction  
+   - Net zero → cancelled out (no flake)
+
+This means that if you assert a value in one commit and retract it in another, they cancel out and disappear from the squashed result.
+
+Future support planned for:
+- Regular 3-way merge (without squash)
+- Schema-aware conflict resolution when validation is added
+
+#### Options
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `ff` | keyword | `:auto` | Fast-forward behavior: `:auto`, `:only`, `:never` |
+| `squash?` | boolean | `false` | Combine all commits into one |
+| `message` | string | auto-generated | Commit message for the new commit |
+| `preview?` | boolean | `false` | Dry run without making changes |
+
+**Note on `:message`:**
+- The `:message` option sets the commit message for the new commit created by the merge
+- If not provided, auto-generates: `"Squash merge from <source-branch>"` or similar
+- This is stored as the commit's message property (separate from annotation)
+- For fast-forward operations, no new commit is created so the message is not used
+
+#### Examples
+
+##### Standard Merge (Auto fast-forward when possible)
+```clojure
+;; Automatically fast-forwards if possible, otherwise fails
+(fluree/merge! conn "ledger:feature" "ledger:main")
+```
+
+##### Squash Merge
+```clojure
+;; Combines all commits from feature branch into single commit on main
+(fluree/merge! conn "ledger:feature" "ledger:main"
+  {:squash? true
+   :message "Feature X implementation"})
+```
+
+##### Fast-forward Only
+```clojure
+(fluree/merge! conn "ledger:feature" "ledger:main"
+  {:ff :only})  ; Fail if fast-forward not possible
+```
+
+#### Response
+
+```clojure
+{:status :success  ; or :error
```

Comment:
I feel like the return should just be the same as a regular commit - return the db with the new HEAD.